sound/soc/sof/intel/hda-loader-skl.c

```diff
@@ -106,30 +106,20 @@
 			(((x) << HDA_CL_SPBFIFO_SPBFCCTL_SPIBE_SHIFT) & \
 			 HDA_CL_SPBFIFO_SPBFCCTL_SPIBE_MASK)
 
-static int cl_skl_cldma_setup_bdle(struct snd_sof_dev *sdev,
-				   struct snd_dma_buffer *dmab_data,
-				   __le32 **bdlp, int size, int with_ioc)
+static void cl_skl_cldma_setup_bdle(struct snd_sof_dev *sdev,
+				    struct snd_dma_buffer *dmab_data,
+				    __le32 **bdlp)
 {
+	const unsigned int bufsize = HDA_SKL_CLDMA_MAX_BUFFER_SIZE;
+	phys_addr_t addr = virt_to_phys(dmab_data->area);
 	__le32 *bdl = *bdlp;
-	int frags = 0;
 
-	while (size > 0) {
-		phys_addr_t addr = virt_to_phys(dmab_data->area +
-						(frags * size));
+	bdl[0] = cpu_to_le32(lower_32_bits(addr));
+	bdl[1] = cpu_to_le32(upper_32_bits(addr));
 
-		bdl[0] = cpu_to_le32(lower_32_bits(addr));
-		bdl[1] = cpu_to_le32(upper_32_bits(addr));
+	bdl[2] = cpu_to_le32(bufsize);
 
-		bdl[2] = cpu_to_le32(size);
-
-		size -= size;
-		bdl[3] = (size || !with_ioc) ? 0 : cpu_to_le32(0x01);
-
-		bdl += 4;
-		frags++;
-	}
-
-	return frags;
+	bdl[3] = cpu_to_le32(0x01);
```

Comment:
can you please add a comment that the DMA uses a single physically-contiguous memory area and that you want an interrupt at the end of the transfer?

Reply:
OK, done.